mapalign/embed.py

```diff
@@ -77,32 +81,8 @@ def compute_diffusion_map(L, alpha=0.5, n_components=None, diffusion_time=0,
             raise ImportError('Checks require scikit-learn, but not found')
 
     ndim = L.shape[0]
-    if overwrite:
-        L_alpha = L
-    else:
-        L_alpha = L.copy()
-
-    if alpha > 0:
-        # Step 2
-        d = np.array(L_alpha.sum(axis=1)).flatten()
-        d_alpha = np.power(d, -alpha)
-        if use_sparse:
-            L_alpha.data *= d_alpha[L_alpha.indices]
-            L_alpha = sps.csr_matrix(L_alpha.transpose().toarray())
-            L_alpha.data *= d_alpha[L_alpha.indices]
-            L_alpha = sps.csr_matrix(L_alpha.transpose().toarray())
-        else:
-            L_alpha = d_alpha[:, np.newaxis] * L_alpha 
-            L_alpha = L_alpha * d_alpha[np.newaxis, :]
 
-    # Step 3
-    d_alpha = np.power(np.array(L_alpha.sum(axis=1)).flatten(), -1)
-    if use_sparse:
-        L_alpha.data *= d_alpha[L_alpha.indices]
-    else:
-        L_alpha = d_alpha[:, np.newaxis] * L_alpha
-
-    M = L_alpha
+    M = _compute_markov_matrix(L, use_sparse = use_sparse, alpha = alpha)
```

Comment:
`M = _compute_markov_matrix(L if overwrite else L.copy(), use_sparse = use_sparse, alpha = alpha)`

Reply:
Regarding this comment and overwrite - I agree that it would be nice to have a memory saving option, but I am not sure if this refactor will allow to manipulate that. I don't know the details of the lexical scope of Python, but maybe you could enlighten me. _compute_markov_matrix creates a local binding A. If we pass an alias as an argument (L) rather than the object itself (L.copy()), is A also going to be an alias? In other words, if we pass an alias to variable that is in the lexical scope of function compute_diffusion_map as an argument to function _compute_markov_matrix,  is _compute_markov_matrix going to mutate the object that the alias refers to OR create it's local copy and mutate that instead?

Sorry for slightly confusing description, but I don't know how to phrase this in a simpler way.